From a872dc940f6322af875713960d6c72fe0f916e4f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Karel=20Ko=C4=8D=C3=AD?= <karel.koci@nic.cz>
Date: Wed, 20 Feb 2019 12:51:05 +0100
Subject: [PATCH] python,python3: do not package binary files to base packages
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Thanks to fix 200a5a2eec896b2d34f1114d14df66d810c1f4bd all base packages
now contain all binaries that are generated as part of python
installation. That causes collision between those packages with package
managers that consider this such as Turris updater-ng. This is also just
wrong. Those binaries were not included and should not be after
mentioned fix as well.

This just adds empty install definition. The idea is to override the
default one that is otherwise used.

Signed-off-by: Karel Kočí <karel.koci@nic.cz>
---
 lang/python/python/Makefile  | 5 +++++
 lang/python/python3/Makefile | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/lang/python/python/Makefile b/lang/python/python/Makefile
index 041b3028c..ff4e16590 100644
--- a/lang/python/python/Makefile
+++ b/lang/python/python/Makefile
@@ -106,6 +106,11 @@ define PyBasePackage
       -|/usr/lib/python$(PYTHON_VERSION)/*/tests
     endif
   endef
+  ifndef PyPackage/$(1)/install
+    define PyPackage/$(1)/install
+		:
+    endef
+  endif
 endef
 
 include ./files/python-package-*.mk
diff --git a/lang/python/python3/Makefile b/lang/python/python3/Makefile
index a88b8165c..3161b3908 100644
--- a/lang/python/python3/Makefile
+++ b/lang/python/python3/Makefile
@@ -110,6 +110,7 @@ define Py3BasePackage
       -|/usr/lib/python$(PYTHON_VERSION)/*/tests
     endif
   endef
+  PyPackage/$(1)/install?=:
 endef
 
 include ./files/python3-package-*.mk
@@ -264,6 +265,9 @@ define Py3Package/python3-base/install
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libpython$(PYTHON_VERSION).so* $(1)/usr/lib/
 endef
 
+PyPackage/python3-light/install?=:
+PyPackage/python3/install?=:
+
 define Py3Package/python3/filespec
 -|$(PYTHON3_PKG_DIR)
 endef
-- 
2.20.1

